src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -723,6 +723,11 @@ public static bool CheckConstraints(this NamedTypeSymbol type, in CheckConstrain
         // metadata and be instantiated in C#.  We check to see if that's happened.
         private static bool HasDuplicateInterfaces(NamedTypeSymbol type, ConsList<TypeSymbol> basesBeingResolved)
         {
+            if (type.OriginalDefinition is SourceNamedTypeSymbol)
```

Comment:
The `HasDuplicateInterfaces` method is used for checking duplicate interface in *metadata* symbols, so we can ignore source symbols - they cannot have invalid duplicate interfaces anyway - but they can have duplicate interface that differ in nullability which is just a warning in C#, so allowed.

This is what fixes the "type is unsupported by the language" error in https://github.com/dotnet/roslyn/issues/40538.